cpp/tests/copying/shift_tests.cu

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <limits>
+#include <memory>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/cudf_gtest.hpp>
+#include <tests/utilities/table_utilities.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/scalar/scalar.hpp>
+
+using cudf::test::fixed_width_column_wrapper;
+using TestTypes = cudf::test::Types<int32_t>;
+
+template<typename T,
+         typename ScalarType = cudf::experimental::scalar_type_t<T>>
+std::unique_ptr<cudf::scalar>
+make_scalar(
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource()) {
+      auto s = new ScalarType(0, false, stream, mr);
+      return std::unique_ptr<cudf::scalar>(s);
+}
+
+template<typename T,
+         typename ScalarType = cudf::experimental::scalar_type_t<T>>
+std::unique_ptr<cudf::scalar>
+make_scalar(
+    T value,
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource()) {
+      auto s = new ScalarType(value, true, stream, mr);
+      return std::unique_ptr<cudf::scalar>(s);
+}
+
+template<typename T>
+auto lowest = std::numeric_limits<T>::lowest();
+
+template<typename T>
+auto highest = std::numeric_limits<T>::max();
+
+template <typename T>
+struct ShiftTest : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(ShiftTest, cudf::test::FixedWidthTypes);
+
+TYPED_TEST(ShiftTest, OneColumnEmpty)
+{
+    using T =  TypeParam;
+
+    fixed_width_column_wrapper<T> a{};
+
+    cudf::table_view input{ { a } };
+    std::vector<std::unique_ptr<cudf::scalar>> fills{};
+    fills.push_back(make_scalar<T>());
+    cudf::experimental::shift(input, 5, fills);
+}
+
+TYPED_TEST(ShiftTest, MisMatchFillValueCount)
```

Comment:
Isn't this the same as the `OneColumnEmpty` test? The name sounds like you meant to check for an error when number of columns != number of scalars.

Reply:
Yes. This was not implemented correctly / at all. Thank you for catching that.